### DISCUSSION TITLE
How to use flows

### DISCUSSION THREAD

**Discussion by sr258** [goauthentik/authentik#4306] [Q&A]
I want to use authentik to decouple user management from my actual application. I don't want to manage multiple applications with authentik, but only create and manage users for one. (Side question: Is it an overkill to use authentik for this purpose?)

I need some basic functionality that my users can do on their own on a self-service site:

- Register new accounts
- Update account data
- Reset password
- Delete their accounts

Authentik seems to do all of this with flows and the example flows from the documentation seem to do what I want (mostly). What I don't understand is how to use the flows. The URLs you get in the admin console are very verbose and don't like like "production" ready. How can I use simple URLs like `http://my.site/register` and not `http://my.site/if/flow/default-source-enrollment/?next=%2Fif%2Fadmin%2F%23%23%2Fflow%2Fflows%2Fdefault-source-enrollment%3B%257B%2522page%2522%253A%2522page-overview%2522%257D`?

I'd also like to customize the look and feel of the sites. I haven't found a way to do this so far. Is this possible?

**Comment by electrofloat** [comment]
Usually you don't have to use those `http://*/if/flow/default...` links. You only have to use the main authentik link which is usually something like: `https://auth.example.com` (of course you are the one who decides what that domain/subdomain name will be).
Then in authentik you create or modify the `default-authentication-identification` flow to have an Enrollment flow. This makes it so that the main login site will have a Sign up link, where your new users can sign up.
Then when they login they will be able to change password/email/etc.

You can also enable the Recovery flow so that the login page will also have a forgot password link.

On how to use flows.. yeah.. the docs would need a big overhaul on usage.

**Comment by tanberry** [comment]
HI all, I'm the tech writer here at authentik, and this is a great discussion. There was recently an Issue about how to remove the Settings icon from the user interface, and I have started learning more about that. But Yes, I want to add new docs around customizing the sites. And... looks like we need some procedural docs on how to use the flows. 

I can make a start on those; am wondering if you'd be interested in co-drafting this with me, @electrofloat? That would be amazing (and who knows some swag might appear in your mailbox as well). :-)

**Reply by electrofloat** [reply]
Hey!

There's definitely a need for a docs upgrade for authentik and I'm glad a dedicated person is there for that job now. A ton of good settings/tweaks are just not documented at all, and sometimes - if you are persistent enough - you could find a peace of info in some random issue comment here, or some random reddit post, or discord chat.

Also - on flows - a lot of crucial info is missing. Like the above that when you set an enrollment flow a "Sign Up" link magically appears on the login page. Same for forgot password, etc. Or like if you have multiple flows with the same designation which will be that one that is selected (Maybe this is documented already since I remember reading about this, but searching for it now, it is definitely not on the Flows page. Edit: I've just found it, it is on the Tenants page :/)

I'm not sure I'm the right person to co-draft this. I'm myself just started learning authentik and sometimes I catch myself looking at the code itself since I cannot figure out something otherwise. But unfortunately I only understand the python part since I don't "speak" Go or TypeScript at all.

Anyways If I have time I could read the draft and make suggestions, but I cannot promise anything. (also freetime is a bit scarce...)

**Reply by tanberry** [reply]
Thanks so much @electrofloat, great info in here, and thanks for the specifics about what we can clarify (or make easier to find!) in the docs. I'll create an issue, and start working on it all, and when there is a PR I'll for sure tag you on it to see if you have time for a review.  I'm really enjoying learning authentik myself; stay in touch and have fun!
